### ISSUE TITLE
Rule expansion proposal: Add "arrow" option to `func-style`

### ISSUE THREAD

**Issue by jrajav** [eslint/eslint#6330]
Eslint version: 2.11.1

The [func-style](http://eslint.org/docs/rules/func-style) rule only supports enforcing function declarations and function expressions. Due to their terser, cleaner syntax and lack of features encouraging bad practices such as `this` and `arguments`, we prefer to use arrow functions wherever we can. However, there doesn't seem to be any way to enforce this with eslint right now.

An additional string option `"arrow"` could be added alongside `"expression"` and `"declaration"` which would result in these patterns being flagged as incorrect:

```
var foo = function() {
    // ...
};

function foo() {
    // ...
}
```

And this not flagged, being correct:

```
var foo = () => {};
```

**Comment by platinumazure** [comment]
Not quite what you're asking for, but have you looked at [`prefer-arrow-callback`](http://eslint.org/docs/rules/prefer-arrow-callback)?

**Comment by ilyavolodin** [comment]
@jrajav Could you update first post to use our change request template: http://eslint.org/docs/developer-guide/contributing/changes
As far as I understand it, you are asking for enhancement to `func-style` rule, but it really helps us if you use the template, so we don't have to clarify information that we require to evaluate the request.

**Comment by jrajav** [comment]
@ilyavolodin I tried to include everything I saw there - I added the eslint version though. Is there anything still missing that I can elaborate further on?

**Comment by jrajav** [comment]
@platinumazure Yes I have, thank you! That rule only flags functions being passed as parameters (that is, callbacks), though, and not the more common use case of functions being defined and used directly.

**Comment by alberto** [comment]
Thanks for your interest in improving eslint. Unfortunately, it looks like consensus couldn't be reached on this issue and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after 21 days tend never to reach consensus, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.

**Comment by jrajav** [comment]
Did anyone disagree with this? I wasn't aware there was a lack of consensus, only a lack of input... I thought this was a pretty understandable and common-sense gap in the current rule.

**Comment by alberto** [comment]
Sorry, I meant it didn't gain enough support from the core team to be accepted.
